core/src/br/cefetmg/games/minigames/ToothRunner.java

```diff
@@ -0,0 +1,247 @@
+                            dente.chao = false;
+                            pulo.play();
+                            dente.tempoAnimacao = 0;
+                        }else {
+                            dente.sobe = -15;
+                        }
+
+                        return false;
+                    }
+
+                    @Override
+                    public boolean touchUp(int screenX, int screenY, int pointer, int button) {
+                        return false;
+                    }
+
+                    @Override
+                    public boolean touchDragged(int screenX, int screenY, int pointer) {
+                        return false;
+                    }
+
+                    @Override
+                    public boolean mouseMoved(int screenX, int screenY) { return false; }
+
+                    @Override
+                    public boolean scrolled(int amount) {
+                        return false;
+                    }
+                }
+        );
+    }
+
+    @Override
+    public void onUpdate(float dt) {
+        //atualizacao de logica
+        dente.atual = dente.corre.getKeyFrame(dente.tempoAnimacao);
+        if (!dente.chao) {
+            if (dente.sprite.getY()+dente.sobe < 100) {
+                dente.sprite.setY(100);
+                dente.chao = true;
+                dente.sobe = dente.impulso;
+            }else {
+                dente.sprite.setY(dente.sprite.getY() + dente.sobe);
+            }
+
+            dente.sobe -= gravidade.y;
+        }
+
+        //atualiza pirulitos
+        for (int i = 0; i < pirulito.size(); i++) {
+            Enemie enemie = pirulito.get(i);
+            enemie.sprite.setX(enemie.sprite.getX() - this.velocidade);
+            if (enemie.sprite.getX() + enemie.sprite.getWidth() < 0) {
+                enemie.sprite.setSize(enemie.sprite.getWidth(), 100 + random.nextInt(30));
+                enemie.sprite.setX(this.screen.viewport.getWorldWidth()+random.nextInt(100) + 350);
+            }
+        }
+
+        //verificar colisao
+        for (int i = 0; i < pirulito.size(); i++) {
+            Enemie enemie = pirulito.get(i);
+            Rectangle aux = dente.sprite.getBoundingRectangle();
+            aux.setSize(90, 90);
+            if (dente.sprite.getBoundingRectangle().overlaps(enemie.sprite.getBoundingRectangle())) {
+                super.challengeFailed();
+            }
+        }
+
+        int aux = (int) dente.tempoAnimacao;
+
+        if (this.dente.chao)
+            dente.tempoAnimacao += Gdx.graphics.getDeltaTime()*7;
+
+        if ((int) dente.tempoAnimacao > aux) {
+            passo.play();
+        }
+    }
+
+    @Override
+    public void onDrawGame() {
+        //controla se perdeu, para que isso nao seja chamado em um intervalo curto do jogo
+
+        this.screen.batch.draw(bg, 0, 0, delta , 0, (int)this.screen.viewport.getWorldWidth(), (int)this.screen.viewport.getWorldHeight()+250);
+        delta += 4;
```

Comment:
Ao colocar `delta += this.velocidade`, os pirulitos passam a estar colocados de acordo com o cenário em vez de parecer que estão em movimento. Há algum motivo para vocês não ter feito assim?

Além disso, há um problema quando o MiniGame é concluído (com vitória ou derrota): o jogador e os pirulitos param, mas o cenário continua mexendo. Isso acontece exatamente por causa desta linha, visto que ela não deveria fazer parte da função `onDrawGame()`, mas da `onUpdate()` (porque ela está atualizando lógica de jogo e não desenhando).